test/e2e/util.go

```diff
@@ -112,7 +111,8 @@ const (
 
 	// How long to try single API calls (like 'get' or 'list'). Used to prevent
 	// transient failures from failing tests.
-	singleCallTimeout = 30 * time.Second
+	// TODO: client should not apply this timeout to Watch calls. Increased from 30s until that is fixed.
+	singleCallTimeout = 5 * time.Minute
```

Comment:
Our e2e tests have watches that last for more than 30 seconds, like some pod startups or pods being ready with long delays.

#23899 covers that this shouldn't be an issue.